### PR TITLE
Test mirror functionality

### PR DESCRIPTION
**Pull request by nelsonjchen**
Built on top of #25. 

Edited the generator script to specify or not specify a base URL mirror.

I don't know if I'm using this right or if I've set the wrong mirror URL somehow. `-b` doesn't work. I haven't debugged this yet though.

### CODE REVIEW

**Comment by miurahr** [comment]
Argument for '-b' option should  be 'http://mirrors.ocf.berkeley.edu/qt/'    instead of  'http://mirrors.ocf.berkeley.edu/qt/online/qtsdkrepository/'

File: generate_azure_pipelines_yml.py [hunk 1]
```diff
@@ -42,6 +43,11 @@ def __init__(self, platform, build_jobs):
     PlatformBuildJobs('windows', windows_build_jobs),
 ]
 
+mirrors = [
+    None,
+    'http://mirrors.ocf.berkeley.edu/qt/online/qtsdkrepository/'
```
Comment (miurahr) [inline generate_azure_pipelines_yml.py]: it should be   'http://mirrors.ocf.berkeley.edu/qt/'

**Comment by miurahr** [comment]
I've just pushed fix. @nelsonjchen could you pull/fetch your repository from github before further modifications?

**Comment by miurahr** [comment]
Now tests are working, but it seems  too much testing the project.
Does @nelsonjchen think? 

Do we run test case such as
1. Windows  5.11.3 desktop  mirror = some
2. Windows  5.11.3  android  mirror = None
etc..
to reduce number of test case a half.

**Comment by nelsonjchen** [comment]
I agree, it's too much. There are some other jobs that could be dropped too.

**Comment by nelsonjchen** [comment]
People building Android apps usually just build on Linux; no Mac or Windows needed. 

If we're testing mirror functionality, I think it can just be a separate job by itself, not parameterized across the current set of jobs.

We can also move lint out to its own job on Linux. Linting's quick, but it would also help cut down that large `steps.yml` file to just building QT and testing.

**Comment by miurahr** [comment]
>We can also move lint out to its own job on Linux. Linting's quick, but it would also help cut down that large steps.yml file to just building QT and testing.

I agree on it. I've already cut it on master branch. lint job is done on Travis-CI where I put a release automation script.

**Comment by miurahr** [comment]
Merged. Thanks a lot!!

**Pull request closed without merging**